src/main/kotlin/no/nav/medlemskap/regler/v1/Hovedregler.kt

```diff
@@ -1,14 +1,15 @@
 package no.nav.medlemskap.regler.v1
 
+import no.nav.medlemskap.domene.Datagrunnlag
 import no.nav.medlemskap.regler.common.*
 
-class Hovedregler(personfakta: Personfakta) {
+class Hovedregler(datagrunnlag: Datagrunnlag) {
 
     private val resultatliste: MutableList<Resultat> = mutableListOf()
-    private val reglerForRegistrerteOpplysninger = ReglerForRegistrerteOpplysninger(personfakta)
-    private val reglerForRegistrerteOpplysningerIMedl = ReglerForMedl(personfakta)
-    private val reglerForGrunnforordningen = ReglerForGrunnforordningen(personfakta)
-    private val reglerForArbeidsforhold = ReglerForArbeidsforhold(personfakta)
+    private val reglerForRegistrerteOpplysninger = ReglerForRegistrerteOpplysninger(datagrunnlag)
```

Comment:
Jeg synes det du gjør er flott, og jeg tror det gjør koden både enklere å lese og forstå. Så jeg er for dette her!

Så lurer jeg på om det kan være et poeng å ta det enda et skritt lenger også? Nå sender du inn datagrunnlag til alle ReglerForXYZ. Men jeg mistenker at en del av dem ikke bruker hele datagrunnlaget, men bare en del av det. Vil det være greit om vi lager en metode som mapper Datagrunnlag til et subsett per regel, så vi får DatagrunnlagForReglerForMedl osv? Eller er det mer jobb enn det gir verdi? Jeg tenker at det vil bli lettere å lage tester hvis ReglerForXYZ bare tar inn de dataene de faktisk trenger, da kan vi opprette de testdataene vi faktisk har behov for, og ikke masse "listefyll" som aldri vil brukes.. Hva tror du?

Reply:
Enig, det passer bra. Spesielt for testene som du sier. Tar dette i egen branch.